### PR TITLE
testing/wireguard-tools: add missing depends

### PR DESCRIPTION
**Pull request by itoffshore**
`wg-quick` does not work with some of the `busybox` built-ins.

reported by Nathan Caldwell (`saintdev at gmail dot com`)

### CODE REVIEW

File: testing/wireguard-tools/APKBUILD [hunk 1]
```diff
@@ -10,6 +10,7 @@ arch='all'
 url='https://www.wireguard.com'
 license="GPL-2.0"
 makedepends="libmnl-dev"
+depends="bash iproute2 procps"
```
Comment (andypost) [inline testing/wireguard-tools/APKBUILD]: does it really require `bash`?

Comment (itoffshore) [reply testing/wireguard-tools/APKBUILD]: `wg-quick` specifies:

```
#!/bin/bash
# SPDX-License-Identifier: GPL-2.0
```

& makes use of `bash` arrays in various places:
```
LXC-alpine [~]$ grep "]}" $(which wg-quick)
SELF="$(readlink -f "${BASH_SOURCE[0]}")"
	INTERFACE="${BASH_REMATCH[2]}"
	[[ $UID == 0 ]] || exec sudo -p "$PROGRAM must be run as root. Please enter the password for %u to continue: " -- "$BASH" -- "$SELF" "${ARGS[@]}"
		output="$(ip route get "${BASH_REMATCH[1]}" || true)"
		[[ ( $output =~ mtu\ ([0-9]+) || ( $output =~ dev\ ([^ ]+) && $(ip link show dev "${BASH_REMATCH[1]}") =~ mtu\ ([0-9]+) ) ) && ${BASH_REMATCH[1]} -gt $mtu ]] && mtu="${BASH_REMATCH[1]}"
		[[ ( $output =~ mtu\ ([0-9]+) || ( $output =~ dev\ ([^ ]+) && $(ip link show dev "${BASH_REMATCH[1]}") =~ mtu\ ([0-9]+) ) ) && ${BASH_REMATCH[1]} -gt $mtu ]] && mtu="${BASH_REMATCH[1]}"
		echo "${BASH_REMATCH[1]}." && return 0
	[[ ${#DNS[@]} -gt 0 ]] || return 0
	printf 'nameserver %s\n' "${DNS[@]}" | cmd resolvconf -a "$(resolvconf_iface_prefix)$INTERFACE" -m 0 -x
	[[ ${#DNS[@]} -gt 0 ]] || return 0
	for address in ${BASH_REMATCH[1]}; do
		[[ $address =~ ^nameserver\ ([a-zA-Z0-9_=+:%.-]+)$ ]] && new_config+="DNS = ${BASH_REMATCH[1]}"$'\n'
	[[ -n $MTU && $(ip link show dev "$INTERFACE") =~ mtu\ ([0-9]+) ]] && new_config+="MTU = ${BASH_REMATCH[1]}"$'\n'
	for cmd in "${PRE_UP[@]}"; do
	for cmd in "${POST_UP[@]}"; do
	for cmd in "${PRE_DOWN[@]}"; do
	for cmd in "${POST_DOWN[@]}"; do
	execute_hooks "${PRE_UP[@]}"
	for i in "${ADDRESSES[@]}"; do
	execute_hooks "${POST_UP[@]}"
	execute_hooks "${PRE_DOWN[@]}"
	execute_hooks "${POST_DOWN[@]}"
```

**Comment by ncopa** [comment]
This was discussed in https://github.com/alpinelinux/aports/pull/3903

I think the better solution would be to to move `wg-quick` to its own package.

@zx2c4 what do you think?

**Comment by zx2c4** [comment]
I certainly don't like the idea of moving wg-quick to its own package. The shebang in wg-quick clearly needs bash; don't try to hack it to use a busybox shebang. Rather, just leave it as is, and if the user doesn't have bash installed, the error running wg-quick will indicate that clearly they need bash.

**Comment by itoffshore** [comment]
I've removed `bash` as a depends & revised the commit message explaining why

**Comment by ncopa** [comment]
> I certainly don't like the idea of moving wg-quick to its own package.

What are the drawbacks? Or what are the benefits in shipping it non-functional with main package?

> Rather, just leave it as is, and if the user doesn't have bash installed, the error running wg-quick will indicate that clearly they need bash.

The error message does not indicate that bash i missing:

````
/ # apk add wireguard-tools
fetch http://dl-cdn.alpinelinux.org/alpine/edge/testing/x86_64/APKINDEX.tar.gz
(1/2) Installing libmnl (1.0.4-r0)
(2/2) Installing wireguard-tools (0.0.20181119-r0)
Executing busybox-1.29.3-r3.trigger
OK: 5 MiB in 15 packages
/ # wg-quick
/bin/sh: wg-quick: not found
/ # ls -l /usr/bin/wg-quick
-rwxr-xr-x    1 root     root         10459 Nov 20 15:34 /usr/bin/wg-quick
````

So I disagree, and would prefer to move wg-quick to a separate subpackage. It does not increase any maintenance work, and for the end user the difference is `apk add wg-quick` instead of `apk add bash` when hitting the _wg-quick: not found_ error.

Something like this:
````diff
diff --git a/testing/wireguard-tools/APKBUILD b/testing/wireguard-tools/APKBUILD
index dfb5c83750..7239d9dcda 100644
--- a/testing/wireguard-tools/APKBUILD
+++ b/testing/wireguard-tools/APKBUILD
@@ -9,8 +9,9 @@ pkgdesc="Next generation secure network tunnel: userspace tools"
 arch='all'
 url='https://www.wireguard.com'
 license="GPL-2.0"
-makedepends="libmnl-dev"
-subpackages="$pkgname-doc $pkgname-bash-completion:bashcomp:noarch"
+makedepends="libmnl-dev bash"
+subpackages="$pkgname-doc $pkgname-bash-completion:bashcomp:noarch
+       wg-quick:wg_quick:noarch"
 options="!check"
 source="https://git.zx2c4.com/WireGuard/snapshot/WireGuard-$pkgver.tar.xz"
 builddir="$srcdir"/WireGuard-$pkgver
@@ -43,4 +44,10 @@ bashcomp() {
        mv "$pkgdir"/usr/share "$subpkgdir"/usr
 }
 
+wg_quick() {
+       depends="bash"
+       mkdir -p "$subpkgdir"/usr/bin
+       mv "$pkgdir"/usr/bin/wg-quick "$subpkgdir"/usr/bin/
+}
+
 sha512sums="dc9286e536bb3f4a6545261b98b5bb056a085f2e95b1b571350168af23cda5231a90b915b03dd29d195412fc61efc781b7d57b2a15a42a34b97e989a5105198e  WireGuard-0.0.20181119.tar.xz"
````

**Comment by zx2c4** [comment]
```diff
diff --git a/src/tools/wg-quick/linux.bash b/src/tools/wg-quick/linux.bash
index 48ce163..a2a0738 100755
--- a/src/tools/wg-quick/linux.bash
+++ b/src/tools/wg-quick/linux.bash
@@ -1,9 +1,18 @@
-#!/bin/bash
+#!/bin/sh
 # SPDX-License-Identifier: GPL-2.0
 #
 # Copyright (C) 2015-2018 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
 #
 
+if [ -z "$BASH_VERSION" ]; then
+	if ! type bash >/dev/null 2>&1; then
+		echo "Please run 'apk add bash' and try again." >&2
+		exit 1
+	fi
+	exec bash "$0" "$@"
+fi
+set +o posix
+
 set -e -o pipefail
 shopt -s extglob
 export LC_ALL=C
```

This will give better error messages.

**Comment by andypost** [comment]
@itoffshore please add patch from previous comment to commit - having this message is needed

**Comment by druilaap** [comment]
wg-quick also depends on a couple more packages. Namely:

- procps for systctl
- openresolv for dns
- iproute2 for /sbin/ip and ipv6 routing

If you do not want to add these dependencies to wireguard-tools, should wg-quick not be patched to to print error messages in wg-quick for those missing dependencies? Currently this leads to hard to parse error messages. e.g. missing command line options for ip, etc. [1]

Cheers,
Druilaap

[1]: https://lists.zx2c4.com/pipermail/wireguard/2018-December/003608.html

**Comment by itoffshore** [comment]
@andypost - Alpine compat patch added

@zx2c4 - added `coreutils` depends as Alpine will use them over the `busybox` builtin tools (for `sysctl -r`)

**Comment by zx2c4** [comment]
That sounds good to me.

**Comment by ncopa** [comment]
you forgot to bump `pkgrel`, but i fixed that for you. Thanks!
